woocommerce/woocommerce-android
# Outdated data top performers

Closes: #11797
Closes: #11798
Closes: #11791

### Why 
As part of the new Background Tasks projects we aim to improve performance of the app by displaying cached data by default.

### Description
This PR aims to enhance the perceived performance of the dashboard UI by leveraging cached data more effectively.

#### Key Changes

Introduce isOutdated Field:

- Added the isOutdated field to top performers state.
- This field helps differentiate between outdated and up-to-date information.

Refreshing Indicator:

- When outdated information is displayed, a refreshing indicator will inform merchants that the data is being updated in the background.

These changes improve the user experience by ensuring that merchants are aware of data updates while benefiting from faster load times due to caching. When the app is refreshing data on the background we present the cached information until the new data arrives.

### Testing information
TC1 

1. Open the app and wait for the dashboards cards to load
2. Close the app and wait 30 min for the cached data to expire
3. Open the app and check that the expired data is displayed 

To skip waiting for 30 min you can run this Patch
```
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt	(revision 9f2048dbe3d6cbf00d40b0fb55c3343674154738)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/sync/AnalyticsUpdateDataStore.kt	(date 1720841339377)
@@ -133,7 +133,7 @@
 
     private fun isElapsedTimeExpired(lastUpdateTime: Long?, maxOutdatedTime: Long): Boolean =
         lastUpdateTime?.let {
-            (currentTime - it) > maxOutdatedTime
+            (currentTime - it) > maxOutdatedTime || true
         } ?: true
 
     private val StatsTimeRangeSelection.identifier

```

TC2
1. Open the app and wait for the dashboards cards to load
2. Pull to refresh the data and check that the stale cached data is displayed while the new data is loading >

### Images/gif

https://github.com/user-attachments/assets/a4b5497c-f456-4cd9-8547-717daa4eb9a7
